anoncreds/src/services/issuer.rs

```diff
@@ -118,9 +115,9 @@ where
 pub fn create_revocation_registry<TW>(
```

Comment:
I believe @TimoGlastra  mentioned in the Slack it should always be `ISSUANCE_BY_DEFAULT`. Do we have to revert this?

Reply:
Yes, I think internally we always to use ISSUANCE_BY_DEFAULT. The property doesn't exist anymore in the spec, so you shouldn't have to provide it when creating a revocation registry. 

The Indy Registry could still use issuance type, but they could just generate the deltas based on it separately: using the full state and you can get the accumulator. Whether you're using issuance on demand or issuance by default the accumulator value will be the same

Reply:
Is it correct that we are no longer supporting issuer using anoncreds to create ISSUANCE_ON_DEMAND revocation registry?

>Whether you're using issuance on demand or issuance by default the accumulator value will be the same

I am not clear about what you mean in the above.

I understand using full state  for supporting indy for subsequent updates after the registry is created.

Reply:
Hmm do you mean the initial revocation list that will be returned by the `anoncreds_create_revocation_registry` method? 

So if I use `ISSUANCE_BY_DEFAULT=true` it will return a list with all `0`s and if I pass `ISSUANCE_BY_DEFAULT=false` it will return all `1`s? I wasn't aware the create_revocation_registry method also returned the first revocation registry entry (which will become the full state right?). In that case the boolean makes sense. 

But it won't be present in the revocation registry definition right, it's just to know how to populate the first revocation registry definition?

Reply:
It creates a few things
- it creates the private key + `RevocationRegistryDefinition` which will not need to contain the issuance type (all good here) 
-  it creates the accumulator value in the` RevocationRegistry`, which is calculated differently for difference issuance type, regardless if we return the `RevocationStatusList`
- it creates the delta (right now) but we need to change it to the `RevocationStatusList` object to return to the issuer to then publish (containing the `RevocationRegistry`

i.e. we are going from:
```
pub fn create_revocation_registry<TW>(
   ...
) -> Result<(
    RevocationRegistryDefinition,
    RevocationRegistryDefinitionPrivate,
    RevocationRegistry,
    RevocationRegistryDelta,
)>
``` 
to
```
pub fn create_revocation_registry<TW>(
   ...
) -> Result<(
    RevocationRegistryDefinition,
    RevocationRegistryDefinitionPrivate,
    RevocationStatusList
)>
``` 

> But it won't be present in the revocation registry definition right, it's just to know how to populate the first revocation registry definition?

yes exactly this, and it is the same for `create_credential`, the issuer needs to know what issuance_type the registry was first calculated with.

Reply:
But the issuer would _only_ need to know it when first creating the revocationList right? Because from then on you can always use the previous revocation list to calculate the new revocation list and accumulator?

In that case it's not exactly this: 
>the issuer needs to know what issuance_type the registry was first calculated with.

But rather:
> The issuer uses the issuance_type to populate the initial `revocationList.revocationList` (the actual array iniside the model with 0/1s). From this revocationList bitarray it can generate the initial accumulator. So if we were to pass in a bit array instead of the `maxCredNum` and `issuanceType`, we would be able to generate the same values. Because the `maxCredNum` can be calculated from the bitarray length, and we don't actually need the issuanceType, because we have the full state.


Sorry if I'm being pedantic here, but I'm really trying to understand what exactly we need to issuanceType for 😊

Reply:
I think it is simply easier for the issuer to pass in the issuance_type here rather than passing in the first ever revocationList and then us working out if there are more 1s or more 0s and assume that the majority will mean that it is the initial issuance type value.